pages/settings/account/index.tsx

```diff
@@ -268,10 +270,14 @@ const AccountSettings = () => {
     }
   })
 
-  const loading = userInfoLoading || status === 'loading'
+  const loading = userInfoLoading || status === SessionStatus.Loading
+  const unauthenticated = status === SessionStatus.unauthenticated
   const { username, name, discordUsername } = data?.userInfo?.user || {}
 
-  if (loading || !data) {
+  useRedirectUnauthenticated(unauthenticated)
+
+  // prevents the page UI from showing if unauthenticated
+  if (loading || unauthenticated) {
```

Comment:
There is a withqueryloader container , we dont have to do this loading logic ourselves
https://github.com/garageScript/c0d3-app/blob/master/containers/withQueryLoader.tsx

Reply:
Looks like it also accounts for error scenarios already too

Reply:
That could be cleaned up in another PR tho

Reply:
Great idea, I'll look into it.